ldap_validator/ldap_validator.py

```diff
@@ -0,0 +1,474 @@
+    ldap_logger.debug("connection entries {}".format(pprint.pformat(entries)))
+    if not(result):
+        ldap_logger.error("connection.result is empty")
+        return False
+    query_rc = int(result['result'])
+    ldap_logger.debug("--------------------------------------------------------------------")
+    ldap_logger.debug("Search query return code: " + str(query_rc))
+    ldap_logger.debug("--------------------------------------------------------------------")
+    ldap_logger.debug("Response: " + str(response['entries']))
+    ldap_logger.debug("--------------------------------------------------------------------")
+    ldap_logger.debug("Result: " + str(result))
+    ldap_logger.debug("--------------------------------------------------------------------")
+    ldap_logger.debug("Entries: ")
+    for entry in entries:
+        results_returned = True
+        ldap_logger.debug(str(entry))
+    ldap_logger.debug("--------------------------------------------------------------------")
+
+    if (verify):
+        try:
+            assert(results_returned is True)
+        except AssertionError:
+            ldap_logger.exception("Error: LDAP Search did not return any connection entries")
+            return False
+    try:
+        assert(query_rc == 0 or query_rc == 4)
+    except AssertionError:
+        ldap_logger.exception("Error: LDAP Search query failed with return code: " + str(query_rc))
+        return False
+
+    ldap_logger.info("LDAP search queries completed successfully")
+    return True
+
+
+# --------------------------------------------------------------------------------------------------
+###################
+#     usage()     #
+###################
+def usage(exit_code: int):
+    """
+    Prints the usage information for the pre-install-report command and exits the program with the given exit_code.
+
+    :param exit_code: The exit code to return when exiting the program.
+    """
+    print("\n")
+    print("Usage: python3 ldap_validator.py <-s|--sitedefault> [<options>]\n")
+    print("Example: python3 ldap_validator.py -s /path/to/sitedefault.yaml\n")
+
+    print("Options:\n")
+    print("    -s  --sitedefault           (Required)Full path the sitedefault.yaml file being validated.")
+    print("    -o, --output-dir=\"<dir>\"    (Optional)Write the report and log files to the provided directory")
+    print("    -d, --debug                 (Optional)Enables logging at DEBUG level. Default is INFO level")
+    print("    -h  --help                  (Optional)Show this usage message")
+    sys.exit(0)
+    print()
+    sys.exit(exit_code)
+
+
+# --------------------------------------------------------------------------------------------------
+##################
+#     main()     #
+##################
+def main(argv):
+
+    """
+    Main execution point for the ldap_validator script
+
+    Argument:
+        arguments(string) - The parameters passed to the script at execution.
+    """
+    global sas_logger
```

Comment:
sas_logger again.    shouldn't this be global declaration of ldap_logger?    And if declared globally, we should be able to remove ldap_logger as input parameter to all methods.

Reply:
So. yes it appears this is the better way and this is how I initially set it up, but it does not work for the Pytest tests.

Reply:
Understood from explanation in review meeting.   declaration of ldap_logger as global was what you tried initially and it doesn't work, so ignore my questions/suggestion here.